Provider fails with panic

Hello,
I've encountered an issue during the plan process: sometimes it panics. It could accidentally happen two or more times in a row, and sometimes just once. And I also see nothing in the controller logs.

Panic happens both on OpenTofu v1.11.4 and Terraform v1.14.4. Lowering the -parallelism value is not helping too, even if I set it to 1.

Provider stacktrace:
```
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: panic: runtime error: invalid memory address or nil pointer dereference
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: [signal SIGSEGV: segmentation violation code=0x2 addr=0x10 pc=0x103112694]
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: goroutine 115 [running]:
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: terraform-provider-ziti/internal/provider.ReadZitiResource({0x1400003e1e0, 0x4e}, {0x14000040b41, 0x24})
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	terraform-provider-ziti/internal/provider/common.go:56 +0x344
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: terraform-provider-ziti/internal/provider.(*interceptV1ConfigResource).Read(0x140008b0008, {0x1037abb88, 0x14000382420}, {{{{0x1037b1370, 0x140004fe510}, {0x10366e440, 0x140004fe120}}, {0x1037b3b10, 0x1400033ebe0}}, 0x140004b8000, ...}, ...)
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	terraform-provider-ziti/internal/provider/config_intercept_v1_resource.go:362 +0x1c0
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x140003b4b48, {0x1037abb88, 0x14000382420}, 0x140003824e0, 0x140004c35a8)
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	github.com/hashicorp/terraform-plugin-framework@v1.14.0/internal/fwserver/server_readresource.go:119 +0x6a0
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0x140003b4b48, {0x1037abb88?, 0x140003822d0?}, 0x14000131dc0)
2026-02-09T13:26:01.195+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	github.com/hashicorp/terraform-plugin-framework@v1.14.0/internal/proto6server/server_readresource.go:55 +0x2e0
2026-02-09T13:26:01.196+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0x1400033db80, {0x1037abb88?, 0x140005c3d40?}, 0x140002235e0)
2026-02-09T13:26:01.196+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/tf6server/server.go:785 +0x228
2026-02-09T13:26:01.196+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x10377bd00, 0x1400033db80}, {0x1037abb88, 0x140005c3d40}, 0x14000380b80, 0x0)
2026-02-09T13:26:01.196+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	github.com/hashicorp/terraform-plugin-go@v0.26.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:575 +0x1c0
2026-02-09T13:26:01.196+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: google.golang.org/grpc.(*Server).processUnaryRPC(0x1400028ac00, {0x1037abb88, 0x140005c3ce0}, 0x1400009c2a0, 0x1400047b7a0, 0x103dcf648, 0x0)
2026-02-09T13:26:01.196+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	google.golang.org/grpc@v1.69.4/server.go:1392 +0xc58
2026-02-09T13:26:01.196+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: google.golang.org/grpc.(*Server).handleStream(0x1400028ac00, {0x1037ac480, 0x140002f8000}, 0x1400009c2a0)
2026-02-09T13:26:01.197+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	google.golang.org/grpc@v1.69.4/server.go:1802 +0x910
2026-02-09T13:26:01.197+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: google.golang.org/grpc.(*Server).serveStreams.func2.1()
2026-02-09T13:26:01.197+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	google.golang.org/grpc@v1.69.4/server.go:1030 +0x8c
2026-02-09T13:26:01.197+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 11
2026-02-09T13:26:01.197+0100 [ERROR] provider.terraform-provider-ziti_v1.0.0: 	google.golang.org/grpc@v1.69.4/server.go:1041 +0x13c
2026-02-09T13:26:01.200+0100 [ERROR] provider: plugin process exited: plugin=.terraform/providers/registry.opentofu.org/netfoundry/ziti/1.0.0/darwin_arm64/terraform-provider-ziti_v1.0.0 id=50044 error="exit status 2"
```

--- a/changelogs/CHANGELOG.1.0.md
+++ b/changelogs/CHANGELOG.1.0.md
@@ -1,3 +1,11 @@
+# v1.0.3
+
+## What's New
+
+* Fix panic error in the common functions logs
+
+    * [Issue #32](https://github.com/netfoundry/terraform-provider-ziti/issues/32) - Provider fails with panic
+
 # v1.0.2
 
 ## What's New

--- a/internal/provider/common.go
+++ b/internal/provider/common.go
@@ -21,7 +21,7 @@ func CreateZitiResource(authUrl string, sessionToken string, payloadData []byte)
 	resp, err := httpClient.Do(req)
 
 	if err != nil {
-		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %v, Status Code: %v, ERR: %v", authUrl, resp.StatusCode, err)
+		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %s, ERR: %v", authUrl, err)
 		return "", err
 	}
 
@@ -53,7 +53,7 @@ func ReadZitiResource(authUrl string, sessionToken string) (string, error) {
 	resp, err := httpClient.Do(req)
 
 	if err != nil {
-		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %v, Status Code: %v, ERR: %v", authUrl, resp.StatusCode, err)
+		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %s, ERR: %v", authUrl, err)
 		return "", err
 	}
 
@@ -85,7 +85,7 @@ func UpdateZitiResource(authUrl string, sessionToken string, payloadData []byte)
 	resp, err := httpClient.Do(req)
 
 	if err != nil {
-		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %v, Status Code: %v, ERR: %v", authUrl, resp.StatusCode, err)
+		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %s, ERR: %v", authUrl, err)
 		return "", err
 	}
 
@@ -117,7 +117,7 @@ func PatchZitiResource(authUrl string, sessionToken string, payloadData []byte) 
 	resp, err := httpClient.Do(req)
 
 	if err != nil {
-		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %v, Status Code: %v, ERR: %v", authUrl, resp.StatusCode, err)
+		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %s, ERR: %v", authUrl, err)
 		return "", err
 	}
 
@@ -149,7 +149,7 @@ func DeleteZitiResource(authUrl string, sessionToken string) (string, error) {
 	resp, err := httpClient.Do(req)
 
 	if err != nil {
-		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %v, Status Code: %v, ERR: %v", authUrl, resp.StatusCode, err)
+		log.Error().Msgf("Got an error back from the Ziti controller: resourceType: %s, ERR: %v", authUrl, err)
 		return "", err
 	}
 

